test/python/transpiler/test_instruction_alignments.py

```diff
@@ -0,0 +1,314 @@
+        instruction_durations.update(
+            [
+                ("rz", (0,), 0),
+                ("rz", (1,), 0),
+                ("x", (0,), 160),
+                ("x", (1,), 160),
+                ("sx", (0,), 160),
+                ("sx", (1,), 160),
+                ("cx", (0, 1), 800),
+                ("cx", (1, 0), 800),
+                ("measure", (0,), 1600),
+                ("measure", (1,), 1600),
+            ]
+        )
+        self.time_conversion_pass = TimeUnitConversion(inst_durations=instruction_durations)
+        self.scheduling_pass = ALAPSchedule(durations=instruction_durations)
+        self.align_measure_pass = AlignMeasures(alignment=16)
+
+    def test_t1_experiment_type(self):
+        """Test T1 experiment type circuit.
+
+        (input)
+
+             ┌───┐┌────────────────┐┌─┐
+        q_0: ┤ X ├┤ Delay(100[dt]) ├┤M├
+             └───┘└────────────────┘└╥┘
+        c: 1/════════════════════════╩═
+                                     0
+
+        (aligned)
+
+             ┌───┐┌────────────────┐┌─┐
+        q_0: ┤ X ├┤ Delay(112[dt]) ├┤M├
+             └───┘└────────────────┘└╥┘
+        c: 1/════════════════════════╩═
+                                     0
+
+        This type of experiment slightly change delay duration of interest.
+        However the quantization error should be less than alignment * dt.
+        """
+        circuit = QuantumCircuit(1, 1)
+        circuit.x(0)
+        circuit.delay(100, 0, unit="dt")
+        circuit.measure(0, 0)
+
+        timed_circuit = self.time_conversion_pass(circuit)
+        scheduled_circuit = self.scheduling_pass(timed_circuit, property_set={"time_unit": "dt"})
+        aligned_circuit = self.align_measure_pass(
+            scheduled_circuit, property_set={"time_unit": "dt"}
+        )
+
+        ref_circuit = QuantumCircuit(1, 1)
+        ref_circuit.x(0)
+        ref_circuit.delay(112, 0, unit="dt")
+        ref_circuit.measure(0, 0)
+
+        self.assertEqual(aligned_circuit, ref_circuit)
+
+    def test_hanh_echo_experiment_type(self):
+        """Test Hahn echo experiment type circuit.
+
+        (input)
+
+             ┌────┐┌────────────────┐┌───┐┌────────────────┐┌────┐┌─┐
+        q_0: ┤ √X ├┤ Delay(100[dt]) ├┤ X ├┤ Delay(100[dt]) ├┤ √X ├┤M├
+             └────┘└────────────────┘└───┘└────────────────┘└────┘└╥┘
+        c: 1/══════════════════════════════════════════════════════╩═
+                                                                   0
+
+        (output)
+
+             ┌────┐┌────────────────┐┌───┐┌────────────────┐┌────┐┌──────────────┐┌─┐
+        q_0: ┤ √X ├┤ Delay(100[dt]) ├┤ X ├┤ Delay(100[dt]) ├┤ √X ├┤ Delay(8[dt]) ├┤M├
+             └────┘└────────────────┘└───┘└────────────────┘└────┘└──────────────┘└╥┘
+        c: 1/══════════════════════════════════════════════════════════════════════╩═
+                                                                                   0
```

Comment:
this is interesting but I don't really understand how the pass decides to insert a delay here, rather than just increase the second `Delay(100[dt])`

Reply:
If the pass find misalignment of measurement, this just insert the delay in between previous instruction and itself. Meaning SX and measure here. Note that delay is also an instruction, therefore if we have Delay(x dt)-Measure then this will become Delay(x + dx dt)-Measure.